zebra-chain/src/amount.rs

```diff
@@ -191,6 +191,54 @@ where
     }
 }
 
+impl<C> std::ops::Mul<i64> for Amount<C>
+where
+    C: Constraint,
+{
+    type Output = Result<Amount<C>>;
+
+    fn mul(self, rhs: i64) -> Self::Output {
+        let value = self.0 * rhs;
+        value.try_into()
+    }
+}
+
+impl<C> std::ops::Mul<Amount<C>> for i64
+where
+    C: Constraint,
+{
+    type Output = Result<Amount<C>>;
+
+    fn mul(self, rhs: Amount<C>) -> Self::Output {
+        let value = rhs.0 * self;
+        value.try_into()
```

Comment:
Let's re-use the `Mul<u64> for Amount<C>` implementation, so they are consistent:

```suggestion
        rhs.mul(self)
```

Reply:
Done as part of https://github.com/ZcashFoundation/zebra/pull/1051/commits/364f70eea3f19be9c35d884d861b5096018aa4fd